### ISSUE TITLE
Shane/KAST red spectra cut off below ~5850 Angstroms?

### ISSUE THREAD

**Issue by rtlow** [pypeit/PypeIt#537]
Dear PypeIt development team,

I am an undergraduate working at UCSD who is using PypeIt to reduce LICK/Kast data for Dr. Burgasser's CoolStar lab. Despite the run_pypeit script failing at the flux calibration stage, PypeIt still output spectra. After looking at these spectra, Dr. Burgasser told me that these all looked strange. The data I was to reduce was from 3 objects:

HD 164595
![screenshot from 2018-10-25 18-27-26](https://user-images.githubusercontent.com/37086221/47766784-d2352d80-dc8d-11e8-981e-6f1433843c6c.png)

J1655+6224
![screenshot from 2018-10-25 18-27-57](https://user-images.githubusercontent.com/37086221/47766803-eed16580-dc8d-11e8-8f76-1c9b0cd0f9da.png)

Feige 110
![screenshot from 2018-10-25 18-25-44](https://user-images.githubusercontent.com/37086221/47766824-001a7200-dc8e-11e8-9b5c-410382d2e166.png)

These were all observed using Shane/KAST red. Dr. Burgasser pointed out that the spectrum for Feige 110 seems to have been cut off around 5850 Angstroms, and that the spectrum for HD 164595 is uncharacteristically noisy. The data in the original .fits files for both of those objects seem to be good, so this may be either an issue with PypeIt stopping early, or an issue with how spectra are extracted. It is concerning to us that spectra for Feige110 is cut off so soon. Since our brown dwarf targets are very red, if PypeIt is not reading data from below a certain wavelength, this will lead to our data being lost.

I hope you can solve my issues. Thank you for your time.

**Comment by rtlow** [comment]
Dr. Burgasser and I have looked over our data, and we think there definitely is an issue with extracting spectra. 

![slit_trace_a_01_aa](https://user-images.githubusercontent.com/37086221/48035249-e5c21780-e117-11e8-859d-ae174af2c676.png)

This is the slit trace QA image that was produced from the reduction. It appears that it is cut off at 500 pixels. 
I have included the original .fits files for this reduction. File r1012.fits one of the flatfield frames, and file r1033.fits is the .fits file for HD 164595. 

[shane_kast_red_data.zip](https://github.com/pypeit/PypeIt/files/2551176/shane_kast_red_data.zip)

Perhaps PypeIt is not taking into account the Shane/KAST images being vertical rather than horizontal? There currently isn't any information in the documentation on how to deal with flipping the images.

I hope you can help me with my issues. Thank you for your time.

**Comment by tiffanyhsyu** [comment]
Hi, I can try taking a look at your data reduction, but would you mind sending me an arc frame? It looks to me that Pypeit has the correct slit orientation -- the edges of the slit, given by the red and green lines, look correct in the image you've attached, but I'll take a look.

If you don't mind, can I also get a raw file of your standard, so I can try to address your issue with fluxing as well (Issue #536)? Thanks!

**Comment by rtlow** [comment]
Thank you for your swift response!

[shane_kast_red_data_1.zip](https://github.com/pypeit/PypeIt/files/2551245/shane_kast_red_data_1.zip)

File r1000.fits is the arc frame, and r1061.fits is the raw file for the standard. 
I have also attached the rest of the QA files here.
[QA.zip](https://github.com/pypeit/PypeIt/files/2551246/QA.zip)

Thank you for your time.

**Comment by tiffanyhsyu** [comment]
Ok wow, I can't even make it as far as you -- I crash on slit tracing. I'm not sure if that has to do with the fact that I'm only using 1 trace frame.

I see what you mean with your trace being cut off though, even though the slit edges are defined, that QA trace frame looks to be a different proportion than your raw flat image.

I'm afraid I don't have a fix right now but I'll keep looking into this..! Thanks.

**Comment by rtlow** [comment]
I can provide more flat frames if that will help.

**Comment by aburgasser** [comment]
could this be due to using a different sized subframe than "standard"? We don't really see how to specify this anywhere in the reduction file.

**Comment by jhennawi** [comment]
Hi All, 

I think the issue here is that the code thinks saturation is 65000 ADU and hence masks the flat as saturated, which results in masking later on down the line. Anyway, it will be fixed in an upcoming release. 

What is your spectroscopic setup? We might be able to add your files to our development suite, which would make them part of the tests we run for PypeIt.

**Comment by aburgasser** [comment]
Hmm, this is KAST/red, and the red flat image only goes up to 24,000 or so, not 63,000. Unless there is a hidden conversion factor here, I don't see how it would have flagged over half of the image (in particular the regions that have no flat on them!). 

We can put up a set off images for darks, flats, flux std, and science target if it helps to spot check on your end (I think Ryan already sent in a subset of these).

To be frank, this code is such a black box that it is hard to know what needs to be tweaked to get this to work. A more transparent, used-input system for setting up particular instrument parameters and settings such saturation would be helpful.

**Comment by aburgasser** [comment]
I've put a zip file of KAST/red data at:
https://drive.google.com/open?id=1_ZxzW54qfeQ60OP2T7sFeF2Gascg82VO

here's the log;
r1000	arc
r1001-1011	bias
r1012-1031	dome flats
r1038	science target
r1042	telluric calibrator (bright star)
r1061	standard (Feige 110)

**Comment by profxj** [comment]
Just downloaded the zip file and ran PypeIt on the data.
I did comment out the standards (those aren't working
at the moment but soon will be again) and half of the flats
(because there were so many and I note HD 168009 was mislabeled
as a flat).  

Ran fine.  

https://docs.google.com/document/d/1By7xCqr3DnwZuLc4dFuVaSxajG4xafkCZjvnWAg3j2E/edit?usp=sharing

Am sending Adam the PypeIt file.

My guess is that you ran the code while including a few 
bad calibration files (e.g. HD 168009 as a pixelflat).
Or that the development done in the last few weeks 
simply fixed things..

As Adam noted, none of the data values are much above 24,000.
